build.assets/Dockerfile

```diff
@@ -167,10 +167,6 @@ ARG PROTOC_VER
 ARG GOGO_PROTO_TAG
 ENV GOGOPROTO_ROOT ${GOPATH}/src/github.com/gogo/protobuf
 
-# Tell prost to use the system protoc instead of building it from source
-ENV PROTOC_NO_VENDOR true
-ENV PROTOC /usr/local/bin/protoc
-
 RUN (export PROTOC_TARBALL=protoc-${PROTOC_VER}-linux-$(if [ "$BUILDARCH" = "amd64" ]; then echo "x86_64"; else echo "aarch_64"; fi).zip && \
```

Comment:
I noticed the equivalent section to this one in the Centos7 Dockerfile got deleted, can we remove this one too?

Reply:
We still need `protoc` in this Dockerfile, which is our buildbox that is used for `make grpc`.

The CentOS 7 Dockerfile was only ever used by CI builds (and not for `make grpc`), so we can get by without `protoc` now that the prost dependency is gone.